### PR TITLE
feat: add list component

### PR DESCRIPTION
**Pull request by DorianMaliszewski**
## Type

- Feature

#### What is expected?

Migrate our List component. Dropped the `card` variant

#### The following changes where made:

1. Add List Folder with all it needs

2. Did UT on the component

3. Add an utils too orderBy

### CODE REVIEW

File: src/components/List/SortIcon.js [hunk 1]
```diff
@@ -0,0 +1,59 @@
+import styled from '@emotion/styled'
+import PropTypes from 'prop-types'
+import React from 'react'
+import Box from '../Box'
+import Icon from '../Icon'
+
+const StyledContainer = styled(Box)`
```
Comment (chambo-e) [inline src/components/List/SortIcon.js:7]: Use a div directly :)

File: src/components/List/mock.js [hunk 2]
```diff
@@ -0,0 +1,12 @@
+export const generateData = count =>
```
Comment (chambo-e) [inline src/components/List/mock.js:1]: To move in tests folder ?

File: src/components/List/index.js [hunk 3]
```diff
@@ -0,0 +1,359 @@
...
+  table: variantTable,
+}
+
+const Body = ({ children, emptyListText, ...props }) => {
+  const { data, idKey, setRowState, rowsState, isLoading } =
+    useContext(ListContext)
+  if (isLoading === true) {
+    return (
+      <Box display="flex" my={2} justifyContent="center" {...props}>
+        <ActivityIndicator size={50} />
```
Comment (philibea) [inline src/components/List/index.js:34]:
```suggestion
        <ActivityIndicator size={50} active />
```

Comment (philibea) [reply src/components/List/index.js:34]: Would it be possible for a List user to change the Loader to default?

Comment (DorianMaliszewski) [reply src/components/List/index.js:34]: Yes, I'll do

**Comment by vincentaudebert** [comment]
Nice accessibility work 👌

**Comment by DorianMaliszewski** [comment]
> I have some issues with select - multi-select
> https://user-images.githubusercontent.com/14060273/121655456-ec1e8a00-ca9e-11eb-96ce-39662057bbec.mov
> 
> @DorianMaliszewski

Will be fixed in another PR

**Review by vincentaudebert — APPROVED** [review]

**Review by chambo-e — APPROVED** [review]

**Review by philibea — APPROVED** [review]

**Pull request merged**